core/common/src/MapLayerSettings.ts

```diff
@@ -349,24 +367,92 @@ export class MapLayerSettings {
     return children;
   }
+
+  /** @internal */
+  public matchesNameAndUrl(name: string, url: string): boolean {
+    return this.name === name && this.url === url;
+  }
+
+  /** @internal */
+  protected static mapTypeName(type: BackgroundMapType) {   // TBD.. Localization.
+    switch (type) {
+      case BackgroundMapType.Aerial:
+        return "Aerial Imagery";
+      default:
+      case BackgroundMapType.Hybrid:
+        return "Aerial Imagery with labels";
+      case BackgroundMapType.Street:
+        return "Streets";
+    }
+  }
+
+  public setCredentials(userName?: string, password?: string) {
+    this.userName = userName;
+    this.password = password;
+  }
 }
+export class ModelMapLayerSettings extends MapLayerSettingsBase {
+  public readonly modelId: Id64String;
+
+  /** @internal */
+  protected constructor(modelId: Id64String,  name: string, visible = true,
+    transparency: number = 0, transparentBackground = true, isBase = false) {
+    super(name, visible, transparency, transparentBackground, isBase);
+    this.modelId = modelId;
+  }
+  /** Construct from JSON, performing validation and applying default values for undefined fields. */
+  public static override fromJSON(json: ModelMapLayerProps): ModelMapLayerSettings {
+    const transparentBackground = (json.transparentBackground === undefined) ? true : json.transparentBackground;
+    return new this(json.modelId, json.name, json.visible, json.transparency, transparentBackground, json.isBase === true);
+  }
+  /** return JSON representation of this MapLayerSettings object */
+  public override toJSON(): ModelMapLayerProps {
+    const props = super.toJSON();
+    return { ... props, modelId: this.modelId };
+  }
+  /** Create a copy of this MapLayerSettings, optionally modifying some of its properties.
+   * @param changedProps JSON representation of the properties to change.
+   * @returns A MapLayerSettings with all of its properties set to match those of `this`, except those explicitly defined in `changedProps`.
+   */
+  public clone(changedProps: Partial<ModelMapLayerProps>): ModelMapLayerSettings {
+    return ModelMapLayerSettings.fromJSON(this.cloneProps(changedProps));
+  }
+  /** @internal */
+  protected override cloneProps(changedProps: Partial<ModelMapLayerProps>): ModelMapLayerProps {
+    const baseProps = super.cloneProps(changedProps);
+    return {... baseProps, modelId: undefined !== changedProps.modelId ? changedProps.modelId : this.modelId };
+  }
+  /** @internal */
+  public override displayMatches(other: MapLayerSettings): boolean {
+    if (other instanceof ImageMapLayerSettings || !super.displayMatches(other))
+      return false;
+
+    return this.modelId === other.modelId;
+  }
+  /** Return true if all sublayers are invisible (always false as model layers do not include sublayers). */
+  public get allSubLayersInvisible(): boolean {
+    return false;
+  }
+}
+
+export type MapLayerSettings = ImageMapLayerSettings | ModelMapLayerSettings;
```

Comment:
@pmconne Sorry I meant this line will lead do instanceof checks.

Reply:
Ideally code that accepts MapLayerSettings won't have to care what specific subtype it's dealing with. I haven't looked at the code.

Reply:
I believe this is the best way to handle this.  Code that deals with map layers generally will not have to care which subtype they are dealing with -- only if they want to do image specific or model specific operations will they need to do instanceof.   I tried adding a "isImageBased" property that just wrapped instanceof but the compiler doesn't seem to handle that.